onnxruntime/hosting/http/session.cc

```diff
@@ -93,31 +93,34 @@ void HttpSession::Send(Msg&& msg) {
 }
 
 template <typename Body, typename Allocator>
-void HttpSession::HandleRequest(boost::beast::http::request<Body, boost::beast::http::basic_fields<Allocator> >&& req) {
+void HttpSession::HandleRequest(http::request<Body, http::basic_fields<Allocator> >&& req) {
   HttpContext context{};
-  context.request = req;
+  context.request = std::move(req);
 
-  std::string path = req.target().to_string();
-  std::string model_name;
-  std::string model_version;
-  std::string action;
+  std::string path = context.request.target().to_string();
+  std::string model_name, model_version, action;
   handler_fn func;
-  http::status status = routes_->ParseUrl(req.method(), path, model_name, model_version, action, func);
+  http::status status = routes_->ParseUrl(context.request.method(), path, model_name, model_version, action, func);
+
+  // TODO: set request id
 
   if (http::status::ok == status && func != nullptr) {
     func(model_name, model_version, action, context);
   } else {
-    http::response<http::string_body> res{status, req.version()};
-    res.set(http::field::server, BOOST_BEAST_VERSION_STRING);
-    res.set(http::field::content_type, "text/plain");
-    res.keep_alive(req.keep_alive());
-    res.body() = std::string("Something failed\n");
-    res.prepare_payload();
-    context.response = res;
+    context.response.result(status);
   }
 
+  PrepareRequestResponse(context);
   return Send(std::move(context.response));
 }
 
+void HttpSession::PrepareRequestResponse(HttpContext& context) {
+  context.response.set(http::field::server, "ONNXRuntime Hosting");
+  context.response.set(http::field::content_type, "application/json");
```

Comment:
It may not always be `application/json`. It is up to the `accept` field and also could be binary if user sends the protobuf message in binary format.

Reply:
Deleted this method and we are now just calling `prepare_payload()` before `Send`. It's going to be up to the user function to set headers.